doc/nrf/device_guides/working_with_nrf/nrf91/nrf9160.rst

```diff
@@ -263,6 +283,131 @@ Building and programming on the command line
+
+   The drop-down text changes to the type of the selected device, with its SEGGER ID below the name.
+   The **Device memory layout** section also changes its name to the device name, and indicates that the device is connected.
+   If the :guilabel:`Auto read memory` option is selected in the **DEVICE** section of the side panel, the memory layout will update.
+   If it is not selected and you wish to see the memory layout, click :guilabel:`Read` in the **DEVICE** section of the side panel.
+
+#. Click :guilabel:`Add file` in the **FILE** section, and select :guilabel:`Browse`.
+
+   .. figure:: images/programmer_addfile_nrf9160dk.png
+      :alt: Programmer - Add file
+
+      Programmer - Add file
+
+#. Navigate to where you extracted the firmware, and select the :file:`mfwnrf9160_<version-number>.zip` file.
+#. Click :guilabel:`Write` in the **DEVICE** section of the side panel.
+
+   .. figure:: images/programmer_write_nrf9160dk.png
+      :alt: Programmer - Write
+
+      Programmer - Write
+
+   The **Modem DFU** window appears.
+
+   .. figure:: images/programmerapp_modemdfu.png
+      :alt: Modem DFU window
+
+      The Modem DFU window
+
+#. Click the :guilabel:`Write` button in the **Modem DFU** window to update the firmware.
+   Do not unplug or turn off the device during this process.
+
+When the update is complete, you see a success message.
+If you update the application firmware now, you can go directly to Step 5 of :ref:`nrf9160_updating_fw_application`.
+
+.. note::
+
+   If you experience problems updating the modem firmware, click :guilabel:`Erase all` in the **DEVICE** section of the side panel and try updating again.
+
+.. _nrf9160_updating_fw_application:
+.. _nrf9160_gs_updating_fw_application:
+
+Updating the application firmware
+---------------------------------
+
+To update the application firmware using the Programmer app, complete the following steps.
+If you experience any problems during the process, press ``Ctrl+R`` (``command+R`` in macOS) to restart the Programmer app and try again.
+
+1. Open the Programmer app.
+#. Make sure the **PROG/DEBUG SW10** switch (**SW5** on DK v0.9.0 and earlier) on the nRF9160 DK is set to **nRF91** or **nRF52** as appropriate for the application or sample you are programming.
+   See the `Device programming section in the nRF9160 DK User Guide`_ for more information.
+
+   For the :ref:`asset_tracker_v2` application, it must be set to **nRF91**.
+
+#. Connect the nRF9160 DK to the computer with a micro-USB cable, and then turn the DK on.
+#. Click :guilabel:`SELECT DEVICE` and select the DK from the drop-down list.
+   You can identify the nRF9160 DK by the fact that it has three COM ports when you expand its entry.
+
+   .. figure:: images/programmer_com_ports.png
+      :alt: Programmer - COM ports
+
+      Programmer - COM ports
+
+   If the three COM ports are not visible, press ``Ctrl+R`` in Windows or ``command+R`` in macOS to restart the Programmer application.
```

Comment:
This should also be updated if you decide to remove the COM port references above.